### DISCUSSION TITLE
offset tiling behavioral change between 7.1.0-43 and 7.1.0-44

### DISCUSSION THREAD

**Discussion by pgajdos** [ImageMagick/ImageMagick#6996] [Development] [answered]
### ImageMagick version

7.1.1-25

### Operating system, version and so on

Linux

### Description

An example taken from https://imagemagick.org/Usage/canvas/
```
magick -tile-offset +20+20 -tile pattern:checkerboard \
          -size 80x80  xc: -draw 'color 30,20 reset'  offset_pattern_fail.gif
```
produces different output for different versions in question:

**7.1.0-43**
![before](https://github.com/ImageMagick/ImageMagick/assets/4067843/61490e28-e760-4b2c-83cb-f5981fa11675)

**7.1.0-44**
![after](https://github.com/ImageMagick/ImageMagick/assets/4067843/74d7159b-c96a-49c2-977a-ee002f66514e)

**7.1.1-25**
![now](https://github.com/ImageMagick/ImageMagick/assets/4067843/c126b680-b40d-41e5-8f42-b472831a715b)

Offending commit:
https://github.com/ImageMagick/ImageMagick/commit/efb742d2dd2951ac1b62b40fac697d8870860e30

I believe I understand correctly that the example command above describes a corner case and the doc actually discourages to use convert command this way in favour of the follow-up command with -size reset. However I would like to ensure that this does not show deeper problem (fx. there is no noticeable offset in the image);  in the opposite case this could be perhaps perceived as a doc inaccuracy report.

**Comment by snibgo** [comment, marked as answer]
The behaviour you saw in 7.1.0-43 was the old "legacy" behaviour, when IM executed options (settings and operations) in a different order to the order they were given in the command line. This caused difficulty because users could not easily predict the outcome of a command.

In v7, most of this behaviour was corrected, and most options were executed in the same order as they were given in the command line. But there were a few "corner cases", which still had the old behaviour. These have been gradually corrected.

The documentation was written when the legacy behaviour still applied, and correctly says this is a problem. It should be modified to state that this legacy behaviour, and hence the problem, no longer apply. In fact, perhaps mentioning the old problem is confusing, and that example should be removed.

As the documentation says, one solution is with `+size`, and another is:
```
magick -tile-offset +20+20  -tile pattern:checkerboard \
  -size 80x80 xc: -draw 'color 30,20 reset' x3.png
```

**Reply by pgajdos** [reply]
Thank you.

**Comment by urban-warrior** [comment]
We corrected the documentation as you recommended.
